### PR TITLE
Feature/dds psm api xtypes refactor

### PR DESCRIPTION
**Pull request by lemunozm**
- Default delegate.
- code style changes.
- API simplifications according to r** manual.
- Implementation constructors.

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/577/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/503/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/573/)

File: include/dds/core/xtypes/detail/MemberType.hpp [hunk 1]
```diff
@@ -82,15 +80,7 @@ class MemberType
 
     void add_annotation(const xtypes::Annotation& annotations)
     {
```
Comment (franava) [inline include/dds/core/xtypes/detail/MemberType.hpp]: lexically, you are only adding one annotation

Comment (franava) [reply include/dds/core/xtypes/detail/MemberType.hpp]:
```suggestion
    {void add_annotation(const xtypes::Annotation& annotation)}
```

File: include/dds/core/xtypes/detail/MemberType.hpp [hunk 2]
```diff
@@ -82,15 +80,7 @@ class MemberType
 
     void add_annotation(const xtypes::Annotation& annotations)
```
Comment (franava) [inline include/dds/core/xtypes/detail/MemberType.hpp]: Lexically, you are adding a single annotation

Comment (franava) [reply include/dds/core/xtypes/detail/MemberType.hpp]:
```suggestion
    void add_annotation(const xtypes::Annotation& annotation)
```

File: include/dds/core/xtypes/detail/MemberType.hpp [hunk 3]
```diff
@@ -82,15 +80,7 @@ class MemberType
...
-        annotations_.push_back(annotations);
-    }
-
-    template<typename AnnotationIter>
-    void annotations(
-            AnnotationIter begin,
-            AnnotationIter end)
-    {
-        annotations_ = std::vector<xtypes::Annotation>(begin, end);
+        annotations_.emplace_back(annotations);
```
Comment (franava) [inline include/dds/core/xtypes/detail/MemberType.hpp]: same as before

Comment (franava) [reply include/dds/core/xtypes/detail/MemberType.hpp]:
```suggestion
        annotations_.emplace_back(annotation);
```

File: include/dds/core/xtypes/detail/Annotation.hpp [hunk 4]
```diff
@@ -29,34 +29,35 @@ namespace detail {
...
+        : Annotation(AnnotationKind::ID_ANNOTATION_TYPE)
+        , id_(annotation.id())
+    {}
 
-    uint32_t id() const noexcept
-    {
-        return id_;
-    }
+    void id(uint32_t id) { id_ = id; }
+    uint32_t id() const { return id_; }
```
Comment (franava) [inline include/dds/core/xtypes/detail/Annotation.hpp]: we'll fight for the right of inline single line setters/getters

File: include/dds/core/xtypes/detail/MemberType.hpp [hunk 5]
```diff
@@ -35,25 +35,19 @@ class MemberType
...
-        name_(name),
-        dt_(dt),
-        ann_()
+            const DynamicType& dynamic_type,
+            Annotation& annotation)
+        : name_(name)
+        , dynamic_type_(dynamic_type)
     {
-        ann_.push_back(a);
+        annotations_.push_back(annotation);
```
Comment (franava) [inline include/dds/core/xtypes/detail/MemberType.hpp]: I thought we said that a class member could have a straight forward shortage of the name.

File: include/dds/core/xtypes/detail/MemberType.hpp (hunk 2 above)
Comment (lemunozm) [reply include/dds/core/xtypes/detail/MemberType.hpp]: Already fixed

File: include/dds/core/xtypes/detail/MemberType.hpp (hunk 3 above)
Comment (lemunozm) [reply include/dds/core/xtypes/detail/MemberType.hpp]: Already fixed

File: include/dds/core/xtypes/detail/MemberType.hpp (hunk 2 above)
Comment (lemunozm) [reply include/dds/core/xtypes/detail/MemberType.hpp]: Also fixed the parameter style

File: include/dds/core/xtypes/detail/Annotation.hpp (hunk 4 above)
Comment (lemunozm) [reply include/dds/core/xtypes/detail/Annotation.hpp]: :+1:

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/584/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/508/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/580/)

**Pull request merged**